### ISSUE TITLE
Dotnet Maui Microsoft.Data.SqlClient Version="5.1.1" Android .Net 7 vs .Net 8

### ISSUE THREAD

**Issue by genifycom** [dotnet/runtime#92612]
### Description

Using .Net 8 with MAUI, Microsoft.Data.SqlClient works with both Windows and Android

Using .Net 7 with MAUI, Microsoft.Data.SqlClient works with Windows but not with Android

### Version with bug

7.0.92

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

All Android versions

### Did you find any workaround?

None

**Comment by marek-safar** [comment]
@steveisok does it look familiar?

**Comment by steveisok** [comment]
I don't believe it has much to do w/ the nuget package. @simonrozsival was this resolved by some of the .net 8 only changes you made?

**Comment by simonrozsival** [comment]
I think the SqlClient now works on Android thanks to the improvements we made to TLS streams in .NET 8. We backported several fixes into .NET 7 but it wasn't possible for all of them. For example the support for `RemoteCertificateValidationCallback`, which I suspect plays a role in establishing the connection to the database in this case, could not be be backported to .NET 7 due to the complexity of the changes (see https://github.com/dotnet/runtime/pull/77386).
